netman/core/switch_sessions.py

```diff
@@ -49,11 +52,31 @@ def open_session(self, switch, session_id):
             raise
 
         self.logger.info("Switch for session {} connected and in transaction mode, storing session".format(session_id))
-        self.sessions[session_id] = switch
+        self.add_session(switch, session_id)
         self._start_timer(session_id)
 
         return session_id
 
+    def add_session(self, switch, session_id):
+        if session_id in self.sessions:
+            raise SessionAlreadyExists(session_id)
+        self.sessions[session_id] = switch
+        try:
+            self.session_storage.add(switch.switch_descriptor, session_id)
+        except NetmanException as e:
+            self.logger.error('Switch for session {} could not be added in '
+                              'SessionStorage: {}'.format(session_id, e))
+
+    def remove_session(self, session_id):
+        if session_id not in self.sessions:
```

Comment:
again if you're here, you passed the "get_switch_for_session" at line 99 that would have thrown that eexception